.github/workflows/minimum.yml

```diff
@@ -21,6 +21,7 @@ jobs:
     - name: Install dependencies
       run: |
           python -m pip install --upgrade pip
-          python -m pip install invoke .[test]
+          python -m pip install Cython==0.29.24
+          python -m pip install invoke .[test] --no-build-isolation
```

Comment:
why do we need cython?

Reply:
For scikit-learn 0.24 it needs Cython version less than 3.0.0 (it goes 0.29 -> 3.0.0 for some reason) otherwise it won't compile. I am fixing it in the build-system to make minimum versions pass

Reply:
That's weird, I don't think we should need to have Cython because when we install scikit-learn it should already be built